salt/modules/aptpkg.py

```diff
@@ -1401,7 +1401,9 @@ def list_pkgs(versions_as_list=False,
         for realpkg, provides in six.iteritems(virtpkgs_all):
             # grep-available returns info on all virtual packages. Ignore any
             # virtual packages that do not have the real package installed.
-            if realpkg in ret['installed']:
+            # _get_virtual() do not use grep-available if HAS_APT is true
+            # so we can skip loop below
+            if not HAS_APT and realpkg in ret['installed']:
```

Comment:
Now if `not HAS_APT` all the `if not removed` case does nothing.
I mean it's better to add the `not HAS_APT` case to the enclosing `if not removed` 11 lines above:
```python
if not removed and not HAS_APT:
```

Reply:
Hello Dmitry,
thanks for your comment and suggestion. It makes sense. 

I've checked that change - it works.  

```
# salt-call pkg.list_pkgs --output=json -l quiet |  jq '."local" |=with_entries(select( .key | test("nginx") ))'
{
  "local": {
    "nginx-core": "1.10.3-0ubuntu0.16.04.2",
    "nginx-common": "1.10.3-0ubuntu0.16.04.2"
  }
}
```